bgpd/bgp_fsm.c

```diff
@@ -1464,6 +1464,11 @@ int bgp_stop(struct peer *peer)
 
 				/* There is no pending EOR message */
 				if (gr_info->eor_required == 0) {
+					if (gr_info->t_select_deferral) {
```

Comment:
I believe we should first stop it, then cleanup?

Reply:
It has been fixed similar way for t_route_select thread info cleanup. 
Also, THREAD_OFF basically sets the t_select_deferral thread pointer to NULL after cancellation & we would loose the reference to thread to free up the thread info.

Reply:
Got it, thanks.